src/bot_driving.py

```diff
@@ -37,13 +55,16 @@ def predict(self, img: np.ndarray) -> np.ndarray:
         
         detections = self.sess.run([self.output_name], {self.input_name: inputs})[0]
         outputs = self.postprocess(detections)
+        
+        self.update_buffer(outputs)
+        steering_signal = self.calculate_steering()
 
-        assert outputs.dtype == np.float32
-        assert outputs.shape == (2,)
-        assert outputs.max() < 1.0
-        assert outputs.min() > -1.0
+        assert steering_signal.dtype == np.float32
+        assert steering_signal.shape == (2,)
+        assert steering_signal.max() < 1.0
+        assert steering_signal.min() > -1.0
 
-        return outputs
+        return steering_signal
```

Comment:
This entire snippet could be moved to `postprocess()`, but I guess it is fine here too.

Reply:
Valid. It was there already so I just left it as is.